modules/outconBidAdapter.js

```diff
@@ -0,0 +1,53 @@
+import {registerBidder} from '../src/adapters/bidderFactory';
+import {config} from '../src/config';
+
+const BIDDER_CODE = 'outcon';
+export const spec = {
+  code: BIDDER_CODE,
+  isBidRequestValid: function(bid) {
+    return !!((bid.params.pod || (bid.params.internalId && bid.params.publisher)) && bid.params.env && bid.params.bidId);
+  },
+  buildRequests: function(validBidRequests) {
+    for (let i = 0; i < validBidRequests.length; i++) {
+      let par = '';
+      let url = '';
+      if (validBidRequests[i].params.pod != undefined) par = 'get?pod=' + validBidRequests[i].params.pod + '&bidId=' + validBidRequests[i].params.bidId;
+      else par = 'get?internalId=' + validBidRequests[i].params.internalId + '&publisher=' + validBidRequests[i].params.publisher + '&bidId=' + validBidRequests[i].params.bidId;
```

Comment:
As a follow-up to our conversation on the `bidId`, the `bidId` value gets created by Prebid.js core.  So if you want to pass into your server request so that it's there in the bid response, you'll want to read it from `validBidRequests[i].bidId` instead of relying on a `bid.param` value that the publisher would have to populate.

With this current setup, the bid response would be rejected by the prebid core logic and display the following warning message:
```
WARNING: Bidder outcon made bid for unknown request ID: 123456789. Ignoring.
```

Reply:
Thanks, I already changed it, tell me if you need another change.